src/segment_kubectl_test.go

```diff
@@ -63,15 +85,127 @@ func TestKubectlSegment(t *testing.T) {
 
 	for _, tc := range cases {
 		args := &kubectlArgs{
-			kubectlExists: tc.KubectlExists,
-			template:      tc.Template,
-			displayError:  tc.DisplayError,
-			context:       tc.Context,
-			namespace:     tc.Namespace,
-			kubectlErr:    tc.KubectlErr,
+			kubectlExists:       tc.KubectlExists,
+			template:            tc.Template,
+			displayError:        tc.DisplayError,
+			kubectlOutContext:   tc.Context,
+			kubectlOutNamespace: tc.Namespace,
+			kubectlOutUser:      tc.User,
+			kubectlOutCluster:   tc.Cluster,
+			kubectlErr:          tc.KubectlErr,
 		}
 		kubectl := bootStrapKubectlTest(args)
 		assert.Equal(t, tc.ExpectedEnabled, kubectl.enabled(), tc.Case)
 		assert.Equal(t, tc.ExpectedString, kubectl.string(), tc.Case)
 	}
 }
+
+func TestKubectlSegmentDefaultKubeConfig(t *testing.T) {
```

Comment:
The above was refactord into the test format, but the tests below weren't. Any pointers as to why not?

Reply:
When I initially though about the additional parameters required in the configuration matrix for the parameterization as well as what I wanted to test I felt extended it would make it unmanageable. For that reason I chose to add separate tests for the new code path and use the existing pattern for the existing ones. I guess now that it's written down it doesn't look that crazy anymore to extend the parameter list. It could be integrated if that is wanted.

Generally speaking I still feel ambiguous towards having a lot of tests in this pattern. As go test does not natively support parameterized tests I find the workflows around what is basically a long for loop in a single test to be quite clunky. E.g. with a separate test I can easily use my IDE (or console) to selectively run and debug a single test. That is not easily possible for the for loop pattern.

Reply:
@hacst that's a discussion we're not having in this repo as there are too many benefits when it comes to this pattern. If I want to debug a single test, I add that case to the top and we're good to go.

Reply:
Fine with me. I will adjust the PR later today.

Reply:
@hacst rereading that I sound like a complete cunt. Wasn't the intention, I love this addition, but when it comes to tests and how they work for me (as I'll be the one here for quite a while) it's an important concept.

Reply:
No worries. Different projects have different preferred coding styles. I am just not yet familiar with the preferences in this project. I definitely know the pains of being a maintainer (props for that by the way 👍 ).

Reply:
Extended previous test pattern with new tests